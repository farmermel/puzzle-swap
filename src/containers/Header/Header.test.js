import React from 'react';
import { shallow } from 'enzyme';
import { Header,
determineLoginButtons,
renderLoggedIn,
handleSignOut,
renderNotLoggedIn,
mapDispatchToProps } from './Header';

describe('Header', () => {
  it('should match snapshot when user is logged in', () => {
    let wrapper = shallow(<Header loggedIn={true}
                                  setLogin={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when user is not logged in', () => {
    let wrapper = shallow(<Header loggedIn={false}
                                  setLogin={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  })

  it.skip('should have more tests', () => {

  })

  //write tests for helper functions
})