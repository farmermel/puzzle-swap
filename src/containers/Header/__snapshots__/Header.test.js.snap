// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot when user is logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={
        Object {
            "uid": "5",
            "username": "case",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          id="header-wrap"
          replace={false}
          to={
                    Object {
                              "pathname": "/",
                            }
          }
>
          <img
                    alt="puzzle"
                    className="puzzleicon"
                    src="puzzle.svg"
          />
          <h1
                    className="app-title"
          >
                    Puzzle Swap
          </h1>
</Link>,
        <div
          className="header-right"
>
          <div
                    className="user-wrap"
          >
                    <p>
                              case
                    </p>
                    <a
                              onClick={[Function]}
                    >
                              Sign out
                    </a>
          </div>
          <Link
                    replace={false}
                    to={
                              Object {
                                        "pathname": "/messages",
                                      }
                    }
          >
                    <button
                              className="messages"
                    >
                              Messages
                    </button>
          </Link>
</div>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <img
              alt="puzzle"
              className="puzzleicon"
              src="puzzle.svg"
/>,
            <h1
              className="app-title"
>
              Puzzle Swap
</h1>,
          ],
          "id": "header-wrap",
          "replace": false,
          "to": Object {
            "pathname": "/",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "puzzle",
              "className": "puzzleicon",
              "src": "puzzle.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Puzzle Swap",
              "className": "app-title",
            },
            "ref": null,
            "rendered": "Puzzle Swap",
            "type": "h1",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="user-wrap"
>
              <p>
                            case
              </p>
              <a
                            onClick={[Function]}
              >
                            Sign out
              </a>
</div>,
            <Link
              replace={false}
              to={
                            Object {
                                          "pathname": "/messages",
                                        }
              }
>
              <button
                            className="messages"
              >
                            Messages
              </button>
</Link>,
          ],
          "className": "header-right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  case
</p>,
                <a
                  onClick={[Function]}
>
                  Sign out
</a>,
              ],
              "className": "user-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "case",
                },
                "ref": null,
                "rendered": "case",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign out",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign out",
                "type": "a",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="messages"
>
                Messages
</button>,
              "replace": false,
              "to": Object {
                "pathname": "/messages",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Messages",
                "className": "messages",
              },
              "ref": null,
              "rendered": "Messages",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            id="header-wrap"
            replace={false}
            to={
                        Object {
                                    "pathname": "/",
                                  }
            }
>
            <img
                        alt="puzzle"
                        className="puzzleicon"
                        src="puzzle.svg"
            />
            <h1
                        className="app-title"
            >
                        Puzzle Swap
            </h1>
</Link>,
          <div
            className="header-right"
>
            <div
                        className="user-wrap"
            >
                        <p>
                                    case
                        </p>
                        <a
                                    onClick={[Function]}
                        >
                                    Sign out
                        </a>
            </div>
            <Link
                        replace={false}
                        to={
                                    Object {
                                                "pathname": "/messages",
                                              }
                        }
            >
                        <button
                                    className="messages"
                        >
                                    Messages
                        </button>
            </Link>
</div>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <img
                alt="puzzle"
                className="puzzleicon"
                src="puzzle.svg"
/>,
              <h1
                className="app-title"
>
                Puzzle Swap
</h1>,
            ],
            "id": "header-wrap",
            "replace": false,
            "to": Object {
              "pathname": "/",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "puzzle",
                "className": "puzzleicon",
                "src": "puzzle.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Puzzle Swap",
                "className": "app-title",
              },
              "ref": null,
              "rendered": "Puzzle Swap",
              "type": "h1",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="user-wrap"
>
                <p>
                                case
                </p>
                <a
                                onClick={[Function]}
                >
                                Sign out
                </a>
</div>,
              <Link
                replace={false}
                to={
                                Object {
                                                "pathname": "/messages",
                                              }
                }
>
                <button
                                className="messages"
                >
                                Messages
                </button>
</Link>,
            ],
            "className": "header-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    case
</p>,
                  <a
                    onClick={[Function]}
>
                    Sign out
</a>,
                ],
                "className": "user-wrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "case",
                  },
                  "ref": null,
                  "rendered": "case",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign out",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Sign out",
                  "type": "a",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="messages"
>
                  Messages
</button>,
                "replace": false,
                "to": Object {
                  "pathname": "/messages",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Messages",
                  "className": "messages",
                },
                "ref": null,
                "rendered": "Messages",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header should match snapshot when user is not logged in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          id="header-wrap"
          replace={false}
          to={
                    Object {
                              "pathname": "/",
                            }
          }
>
          <img
                    alt="puzzle"
                    className="puzzleicon"
                    src="puzzle.svg"
          />
          <h1
                    className="app-title"
          >
                    Puzzle Swap
          </h1>
</Link>,
        <div
          className="header-right"
>
          <div
                    className="login-container"
          >
                    <Link
                              replace={false}
                              to={
                                        Object {
                                                  "pathname": "/login",
                                                }
                              }
                    >
                              <span
                                        className="login"
                              >
                                        Login
                              </span>
                    </Link>
                    <Link
                              replace={false}
                              to={
                                        Object {
                                                  "pathname": "/sign-up",
                                                }
                              }
                    >
                              <span>
                                        Sign up
                              </span>
                    </Link>
          </div>
</div>,
      ],
      "className": "App-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <img
              alt="puzzle"
              className="puzzleicon"
              src="puzzle.svg"
/>,
            <h1
              className="app-title"
>
              Puzzle Swap
</h1>,
          ],
          "id": "header-wrap",
          "replace": false,
          "to": Object {
            "pathname": "/",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "puzzle",
              "className": "puzzleicon",
              "src": "puzzle.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Puzzle Swap",
              "className": "app-title",
            },
            "ref": null,
            "rendered": "Puzzle Swap",
            "type": "h1",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="login-container"
>
            <Link
                        replace={false}
                        to={
                                    Object {
                                                "pathname": "/login",
                                              }
                        }
            >
                        <span
                                    className="login"
                        >
                                    Login
                        </span>
            </Link>
            <Link
                        replace={false}
                        to={
                                    Object {
                                                "pathname": "/sign-up",
                                              }
                        }
            >
                        <span>
                                    Sign up
                        </span>
            </Link>
</div>,
          "className": "header-right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to={
                                Object {
                                                "pathname": "/login",
                                              }
                }
>
                <span
                                className="login"
                >
                                Login
                </span>
</Link>,
              <Link
                replace={false}
                to={
                                Object {
                                                "pathname": "/sign-up",
                                              }
                }
>
                <span>
                                Sign up
                </span>
</Link>,
            ],
            "className": "login-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span
                  className="login"
>
                  Login
</span>,
                "replace": false,
                "to": Object {
                  "pathname": "/login",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "login",
                },
                "ref": null,
                "rendered": "Login",
                "type": "span",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  Sign up
</span>,
                "replace": false,
                "to": Object {
                  "pathname": "/sign-up",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign up",
                },
                "ref": null,
                "rendered": "Sign up",
                "type": "span",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            id="header-wrap"
            replace={false}
            to={
                        Object {
                                    "pathname": "/",
                                  }
            }
>
            <img
                        alt="puzzle"
                        className="puzzleicon"
                        src="puzzle.svg"
            />
            <h1
                        className="app-title"
            >
                        Puzzle Swap
            </h1>
</Link>,
          <div
            className="header-right"
>
            <div
                        className="login-container"
            >
                        <Link
                                    replace={false}
                                    to={
                                                Object {
                                                            "pathname": "/login",
                                                          }
                                    }
                        >
                                    <span
                                                className="login"
                                    >
                                                Login
                                    </span>
                        </Link>
                        <Link
                                    replace={false}
                                    to={
                                                Object {
                                                            "pathname": "/sign-up",
                                                          }
                                    }
                        >
                                    <span>
                                                Sign up
                                    </span>
                        </Link>
            </div>
</div>,
        ],
        "className": "App-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <img
                alt="puzzle"
                className="puzzleicon"
                src="puzzle.svg"
/>,
              <h1
                className="app-title"
>
                Puzzle Swap
</h1>,
            ],
            "id": "header-wrap",
            "replace": false,
            "to": Object {
              "pathname": "/",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "puzzle",
                "className": "puzzleicon",
                "src": "puzzle.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Puzzle Swap",
                "className": "app-title",
              },
              "ref": null,
              "rendered": "Puzzle Swap",
              "type": "h1",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="login-container"
>
              <Link
                            replace={false}
                            to={
                                          Object {
                                                        "pathname": "/login",
                                                      }
                            }
              >
                            <span
                                          className="login"
                            >
                                          Login
                            </span>
              </Link>
              <Link
                            replace={false}
                            to={
                                          Object {
                                                        "pathname": "/sign-up",
                                                      }
                            }
              >
                            <span>
                                          Sign up
                            </span>
              </Link>
</div>,
            "className": "header-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to={
                                    Object {
                                                      "pathname": "/login",
                                                    }
                  }
>
                  <span
                                    className="login"
                  >
                                    Login
                  </span>
</Link>,
                <Link
                  replace={false}
                  to={
                                    Object {
                                                      "pathname": "/sign-up",
                                                    }
                  }
>
                  <span>
                                    Sign up
                  </span>
</Link>,
              ],
              "className": "login-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span
                    className="login"
>
                    Login
</span>,
                  "replace": false,
                  "to": Object {
                    "pathname": "/login",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                    "className": "login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "span",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span>
                    Sign up
</span>,
                  "replace": false,
                  "to": Object {
                    "pathname": "/sign-up",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": "span",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
